pkg/watcher/reconciler/dynamic/dynamic.go

```diff
@@ -156,6 +158,12 @@ func (r *Reconciler) deleteUponCompletion(ctx context.Context, o results.Object)
 		return controller.NewRequeueAfter(requeueAfter)
 	}
 
+	// Verify whether this object matches the provided label selectors
+	if selectors := r.cfg.GetLabelSelector(); !selectors.Matches(labels.Set(o.GetLabels())) {
+		logger.Debugw("Object doesn't match the required label selectors - requeuing to process later", zap.String("results.tekton.dev/label-selectors", selectors.String()))
+		return controller.NewRequeueAfter(1 * time.Minute)
```

Comment:
One minute is pretty fast for a re-queue of a completed PipelineRun/TaskRun, especially for something that likely won't get deleted. 10 minutes is a typical re-list interval for k8s controllers, I would go with that:

```suggestion
		return controller.NewRequeueAfter(10 * time.Minute)
```

Reply:
What do you think about making this value configurable? Then users may decide what fits in their needs. I'm bringing it up because in our case at Nubank, we prefer to collect the PipelineRun/TaskRun as soon as possible because during high peaks, the cluster gets overloaded very quickly with thousands of simultaneous runs.
So, we could add a new flag (requeue_interval) whose default value would be 10m as you've suggested. For more specific scenarios, users may increase or decrease this value.

Reply:
Done @adambkaplan. Do you mind to have a second look?